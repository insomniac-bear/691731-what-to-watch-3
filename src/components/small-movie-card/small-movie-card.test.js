import React from 'react';
import renderer from 'react-test-renderer';
import SmallMovieCard from './small-movie-card.jsx';

const mockFilmData = {
  id: 0,
  filmName: `Name`,
  posterUrl: `img/bohemian-rhapsody.jpg`,
  filmPreview: `Some film url`,
};

it(`Render SmallMovieCard`, () => {
  const tree = renderer
    .create(<SmallMovieCard
      filmData={mockFilmData}
      hoveredElement={jest.mock()}
      onHover={jest.mock()}
      onMouseOut={jest.mock()}
      onCardClickHandler={jest.mock()}
      renderVideoPreview={() => {}}
    />, {
      createNodeMock: () => {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
