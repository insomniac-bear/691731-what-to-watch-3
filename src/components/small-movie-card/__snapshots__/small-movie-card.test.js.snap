// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SmallMovieCard 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      poster="img/bohemian-rhapsody.jpg"
      src="Some film url"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[Function]}
    >
      Name
    </a>
  </h3>
</article>
`;
